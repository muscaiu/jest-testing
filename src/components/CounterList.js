import React, { Component } from 'react'
import Counter from './Counter'

const createRamge = num => Array.from(Array(num).keys())

class CounterList extends Component {
    state = {
        numCounters: 2
    }

    addCounter = () => {
        const { numCounters } = this.state;
        this.setState({ numCounters: numCounters + 1 })
    }

    renderCounters(index) {
        return (
            <li key={index}>
                <Counter />
            </li>
        )
    }
    render() {
        const { numCounters } = this.state
        const countersArray = createRamge(numCounters)

        return (
            <div>
                <button
                    className={'add'}
                    onClick={this.addCounter}
                >
                    add
                </button>
                <ul>
                    {countersArray.map(num => this.renderCounters(num))}
                </ul>
            </div>
        )
    }
}

export default CounterList