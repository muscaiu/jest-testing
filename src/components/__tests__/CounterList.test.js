import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import CounterList from '../CounterList'
import { wrap } from 'module';

describe('CounterList', () => {
    it('should match the snapshot', () => {
        const tree = renderer.create(<CounterList />).toJSON()

        expect(tree).toMatchSnapshot()
    })
    it('should add a counter when the add button is clicked', () => {
        const wrapper = shallow(<CounterList />)
        const counters = wrapper.find('Counter')

        expect(counters.length).toEqual(2)
    })
    it('should add a counter when the add button is clicked', () => {
        const wrapper = shallow(<CounterList />)
        const btnAdd = wrapper.find('button.add')
        btnAdd.simulate('click')
        const counters = wrapper.find('Counter')
        
        expect(counters.length).toEqual(3)
    })
})
