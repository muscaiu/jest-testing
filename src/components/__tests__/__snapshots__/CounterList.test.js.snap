// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterList should match the snapshot 1`] = `
<div>
  <button
    className="add"
    onClick={[Function]}
  >
    add
  </button>
  <ul>
    <li>
      <div>
        <p>
          Current count: 
          0
        </p>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </li>
    <li>
      <div>
        <p>
          Current count: 
          0
        </p>
        <button
          className="increment"
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="decrement"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </li>
  </ul>
</div>
`;
